packages/@aws-cdk/aws-ecs/lib/container-definition.ts

```diff
@@ -606,6 +598,34 @@ export class ContainerDefinition extends CoreConstruct {
     this.environment[name] = value;
   }
 
+  /**
+   * This method adds a Secret to the container.
+   */
+  public addSecret(name: string, secret: Secret) {
+    if (!this.secrets) {
+      this.secrets = [];
+    }
+    if (secret.hasField) {
+      this._referencesSecretJsonField = true;
+    }
+    secret.grantRead(this.taskDefinition.obtainExecutionRole());
+    this.secrets.push({
+      name,
+      valueFrom: secret.arn,
+    });
+  }
+
+  /**
+   * This method returns true if any Secret in this definition references a JSON field.
+   */
+  public referencesSecretJsonField(): boolean {
+    if (this._referencesSecretJsonField) {
+      return true;
+    } else {
+      return false;
+    }
+  }
```

Comment:
Maybe use getter syntax to allow it to be accessed as a property is what needs to be changed:

`public referencesSecretJsonField(): boolean {`

That would allow it to be accessed as `obj.referencesSecretJsonField` thus not breaking API spec.